bosh/plugin.go

```diff
@@ -4,27 +4,64 @@ import (
 	"encoding/json"
 	"log"
 	"net/http"
+
+	"github.com/cloudfoundry/bosh-agent/jobsupervisor/monit"
 )
 
 type plugin struct {
-	report report
+	report          report
+	monitHTTPClient monit.Client
 }
 
-func NewPlugin(hostname, jobSpecPath string) *plugin {
+func NewPlugin(hostname, jobSpecPath, platform string) (*plugin, error) {
 	report := newReport(hostname)
 
 	if jobSpec, err := loadSpec(jobSpecPath); err != nil {
 		log.Printf("error loading job spec from %q: %v\n", jobSpecPath, err)
+		return nil, err
 	} else {
-		report.AddJob(jobSpec)
+		report.SetInstanceSpecData(jobSpec)
+	}
+
+	var monitHTTPClient monit.Client
+	var err error
+
+	if platform == "ubuntu" {
+		monitHTTPClient, err = newMonitHTTPClient()
+		if err != nil {
+			log.Printf("error creating monit HTTP client: %v\n", err)
+			return nil, err
+		}
 	}
 
-	return &plugin{report}
+	return &plugin{report, monitHTTPClient}, nil
 }
 
 func (p *plugin) Report(w http.ResponseWriter, r *http.Request) {
-	if err := json.NewEncoder(w).Encode(p.report); err != nil {
+	err := p.updateReport()
+
+	if err != nil {
+		log.Printf("error updating report: %v\n", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err = json.NewEncoder(w).Encode(p.report); err != nil {
 		log.Printf("error encoding report: %v\n", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Just fyi, here we don't need the `return` since there's nothing coming afterwards.